torch/csrc/jit/python/init.cpp

```diff
@@ -123,6 +126,81 @@ using ::c10::FunctionSchema;
 using caffe2::serialize::PyTorchStreamReader;
 using caffe2::serialize::PyTorchStreamWriter;
 
+class PythonSymbolicIntNode: public c10::SymbolicIntNode {
+public:
+  PythonSymbolicIntNode(py::object pyobj):
+    c10::SymbolicIntNode() {
+      pyobj_ = std::make_shared<c10::SafePyObject>(pyobj.release().ptr(), getPyInterpreter());
+    };
+
+  virtual SymbolicIntNode* wrap(int64_t num) {
+    auto r = getPyObj().attr("wrap")(num);
+    return new PythonSymbolicIntNode(r);
+  }
+
+  virtual bool bool_() {
+    py::gil_scoped_acquire acquire;
+    return py::str(getPyObj().attr("__bool__")()).is(py::str(Py_True));
+    // TODO: test getPyObj().attr("__bool__")().is(py::object(Py_True))
+  }
+
+  virtual int64_t int_() {
+    py::gil_scoped_acquire acquire;
+    return getPyObj().attr("__int__")().cast<int64_t>();
+  }
+
+  virtual std::string str() {
+    py::gil_scoped_acquire acquire;
+    return getPyObj().attr("__str__")().cast<std::string>();
+    // TODO: can we just do getPyObj().cast<string>?
+  }
+
+  virtual SymbolicIntNode* dispatch_common_(const char* fname, SymbolicIntNode* other) {
+    auto pother = dynamic_cast<PythonSymbolicIntNode*>(other);
+    TORCH_CHECK(pother);
+    auto magic_fname = std::string("__") + fname + std::string("__");
```

Comment:
I'd much rather you had taken the magic_fname as argument lol. With a macro you could paste the __ together with a string constant without having to do a string concat every function call (which is wasteful)

Reply:
haha sorry, I was going to do but didn't do it before your review.